src/init.ts

```diff
@@ -78,6 +87,23 @@ function upgrade (options: InitOptions) {
 }
 
 /**
+ * Make a smart guess of the project from either the `package.json` or `bower.json` files.
+ */
+
+function guessProjectName (options: InitOptions) : string {
+  for (let i = 0; i < CONFIG_NAMES.length; i++) {
+    try {
+      const pack = require(join(options.cwd, CONFIG_NAMES[i]))
```

Comment:
For commands that aren't related to start up, I try to keep everything non-blocking. Using `require` is synchronous, so I'd tweak this.